main/Main.hs

```diff
@@ -322,7 +338,46 @@ reReadConfig path refConf = do
   atomicWriteIORef refConf conf
   putStrLn ("Config file reloaded" :: Text)
 
--- Utilitarian functions.
+
+-- | Dump DbStructure schema to JSON
+dumpSchema :: AppConfig -> IO LBS.ByteString
+dumpSchema conf =
+  do
+    Right conn <- C.acquire . toS $ configDbUri conf
+    Right pgVersion <- S.run getPgVersion conn
+    let
+      getDbStructureTransaction =
+        HT.transaction HT.ReadCommitted HT.Read $
+          getDbStructure
+            (toList $ configDbSchemas conf)
+            (configDbExtraSearchPath conf)
+            pgVersion
+            (configDbPreparedStatements conf)
+    Right dbStructure <- S.run getDbStructureTransaction conn
+    C.release conn
+    return $ Aeson.encode dbStructure
+
+
+-- | Print the time taken to run an IO action to stderr with the given printf string
+timeToStderr :: [Char] -> IO a -> IO a
+timeToStderr fmtString a =
```

Comment:
Do we need that? I thought we'd just do something like `cabal v2-exec time -- postgrest --dump-schema`. Is there a disadvantage with that, that I don't see?

Reply:
Intent is to get the time for just the schema cache loading, without the encoding and printing overhead. With the latest commit we should actually achieve that :-) 

The difference is quite significant:

```
[nix-shell:~/Projects/postgrest]$ test/with_tmp_db cabal v2-exec -- bash -c "time postgrest --dump-schema test/io-tests/configs/simple.config > /dev/null"
Loaded schema in 0.011 seconds

real    0m0,073s
user    0m0,050s
sys     0m0,075s
```

This would be useful to monitor the load time in CI, in the past it got out of hand for large schemas without being noticed for a while.

Reply:
Ok with the changes, I can follow your reasoning. Quite a lot of overhead.